src/v/kafka/server/connection_context.h

```diff
@@ -131,6 +131,7 @@ class connection_context final
     // used to pass around some internal state
     struct session_resources {
         ss::lowres_clock::duration backpressure_delay;
+        bool delay_partition_quota_exceeded;
```

Comment:
Maybe this could be cleaner? Ideally i'd like to define `quota_manager::throttle_delay` here, but that would introduce a circular dependency. Moving that class to a header such as `kafka/types.h` was considered, but I didn't want to pollute that file with such a specific class. In the end I went with this specific bool flag which is passed along to the appropriate subsystems